R/derive_param_maxsev_draft.R

```diff
@@ -0,0 +1,107 @@
+#' Title: DERIVE_PARAM_MAXSEV 
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ 
+#' @description                                                                   +
+#' This 'derive_param_maxsev' function is used to derive the maximum severity for +
+#'  each administrative and Systemic events.                                      + 
+#' ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#' @details 
+#' 
+#' Pre_processing:
+#'    
+#' Severity records for Administrative and systemic events will be filtered 
+#' from the input data set and AVAL will be derived in a data_frame [pp] 
+#' 
+#' Maximum severity derivation:   
+#'     
+#' Once the pre-processing is done. The maximum severity records will be derived
+#' from the  AVAL by USUBJID and [by_var]s.
+#' Below variables will be created with the values mentioned below.
+#' [DTYPE = "MAXIMUM", FATESTCD = "MAXSEV" and FATEST = "MAXIMUM SEVERITY".]
+#' 
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#' 
+#' @param dataset 
+#' The Input data frame for this function
+#' required variables for Input data freame :USUBJID, ATPTREF(FATPTREF),
+#' PARCAT2(FASCAT), AVAL(FASTRESN),AVALC(FASTRESC),FAOBJ,FATESTCD,FATEST.
+#' 
+#' @param testcd_sev 
+#' In testcd_sev, Pass the testcd in the quotes ("SEV") which is representing 
+#' the severity records.if you have more than one testcd, pass it as a list in
+#' the [testcd_sev] arugument.  
+#' 
+#' @param by_vars
+#' In by_vars, pass the inside vars() grouping variables to derive the Maximum severity
+#' eg: by_vars= vars(USUBJID,FAOBJ, ATPTREF)
+#' 
+#' 
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#' @return Dataframe which has Maxsev records (with DTYPE as "MAXIMUM", Derived AVAL
+#' and AVALC and some grouping vars and updated FATESTCD and FATEST.)
+#'
+#' @author ARJUN R
+#'
+#' @examples

```

Comment:
Please make sure the examples are fully self-contained as indicated in the [programming strategy](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html): "A fully self-contained example of how to use the function. Self-contained means that, if this code is executed in a new R session, it will run without errors. That means any packages need to be loaded with [library()](https://rdrr.io/r/base/library.html) and any datasets needed either to be created directly inside the example code or loaded using [data()](https://rdrr.io/r/utils/data.html). If a dataset is created in the example, it should be done so using the function [tribble()](https://tibble.tidyverse.org/reference/tribble.html) (specify [library(tibble)](https://tibble.tidyverse.org/) before calling this function). If other functions are called in the example, please specify [library(pkg_name)](https://rdrr.io/r/base/library.html) then refer to the respective function fun() as opposed to the preferred pkg_name::fun() notation as specified in [Unit Test Guidance](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#set-up-the-test-script). Make sure to align columns as this ensures quick code readability."